### PR TITLE
KIALI-2421 deploy dashboards with our public-facing install scripts

### PR DESCRIPTION
**Pull request by jotak**
JIRA: https://issues.jboss.org/browse/KIALI-2421

### CODE REVIEW

**Comment by jotak** [comment]
For the record, @jmazzitelli told me by chat: the script has to handle resources not being present locally, hence downloading from the remote repo, as it can be called from the `getLatestKiali` url

**Comment by jotak** [comment]
@jmazzitelli it's now downloading if cannot find locally

File: deploy/kubernetes/deploy-kiali-to-kubernetes.sh [hunk 1]
```diff
@@ -228,3 +228,23 @@ do
     exit 1
   fi
 done
+
+# Deploy Kiali MonitoringDashboards to kubernetes
+# Note for undeploy script: dashboards are implicitely undeployed when the related CRD is removed
```
Comment (jmazzitelli) [inline deploy/kubernetes/deploy-kiali-to-kubernetes.sh]: trivial, but remove the "e" - spelled "implicitly"

File: deploy/kubernetes/deploy-kiali-to-kubernetes.sh [hunk 2]
```diff
@@ -228,3 +228,23 @@ do
...
+echo "Deploying Kiali dashboards to Kubernetes project ${NAMESPACE}"
```
Comment (jmazzitelli) [inline deploy/kubernetes/deploy-kiali-to-kubernetes.sh]: There is no concept of "project" in k8s, can you replace this "project" with namespace - and do it in the other spot in the code you didn't change? :) We should say deploying to Kubernetes "namespace"

File: deploy/kubernetes/deploy-kiali-to-kubernetes.sh [hunk 3]
```diff
@@ -228,3 +228,23 @@ do
...
+    echo "Using YAML file: ${yaml_path}"
+    cat ${yaml_path} | envsubst | kubectl apply -n ${NAMESPACE} -f -
+  else
+    get_downloader
+    yaml_url="https://raw.githubusercontent.com/kiali/kiali/${VERSION_LABEL}/deploy/dashboards/${dashboard}.yaml"
+    echo "Downloading YAML via: ${downloader} ${yaml_url}"
+    ${downloader} ${yaml_url} | envsubst | kubectl apply -n ${NAMESPACE} -f -
+  fi
+  if [ "$?" != "0" ]; then
+    echo "WARNING: Failed to deploy runtimes dashboards. If you need them, please deploy manually."
```
Comment (jmazzitelli) [inline deploy/kubernetes/deploy-kiali-to-kubernetes.sh]: Is there anything else we can say here to indicate this isn't really a problem that will break Kiali, that this is an optional feature. I see it is WARNING not ERROR. But I just have a feeling we need to be more explicit to tell the user something to the effect of "you don't really need these to run the base Kiali ...these are just for the optional dashboards." I don't know a good way to say it, but you know what I mean. :)

File: deploy/openshift/deploy-kiali-to-openshift.sh [hunk 4]
```diff
@@ -229,3 +229,23 @@ do
...
+    echo "Using YAML file: ${yaml_path}"
+    cat ${yaml_path} | envsubst | ${OC_TOOL_PATH} apply -n ${NAMESPACE} -f -
+  else
+    get_downloader
+    yaml_url="https://raw.githubusercontent.com/kiali/kiali/${VERSION_LABEL}/deploy/dashboards/${dashboard}.yaml"
+    echo "Downloading YAML via: ${downloader} ${yaml_url}"
+    ${downloader} ${yaml_url} | envsubst | ${OC_TOOL_PATH} apply -n ${NAMESPACE} -f -
+  fi
+  if [ "$?" != "0" ]; then
+    echo "WARNING: Failed to deploy runtimes dashboards. If you need them, please deploy manually."
```
Comment (jmazzitelli) [inline deploy/openshift/deploy-kiali-to-openshift.sh]: same comments as above - "implicitly" and see what other way we can word this WARNING message to make it clear what failed and that it doesn't break base Kiali functionality.

**Comment by jmazzitelli** [comment]
I  changed the title - these are not merely hack scripts - these are our very public-facing install scripts.

**Comment by jmazzitelli** [comment]
I am making @gbaufake a reviewer only because we need to make sure QE tests these scripts, since they are very important.

File: deploy/kubernetes/deploy-kiali-to-kubernetes.sh (hunk 3 above)
Comment (jotak) [reply deploy/kubernetes/deploy-kiali-to-kubernetes.sh]: How about: "WARNING: Failed to deploy runtimes dashboards. They are not mandatory and won't prevent Kiali to work. If you want to monitor your application runtimes, you can still deploy dashboards manually."

Ideally I would add a link to the documentation, but I don't have it yet

**Comment by rhqci** [comment]
Jenkins CI: kiali-core-pr-e2e-test [#598](https://paste.fedoraproject.org/paste/co92zrNcv4WBgE1eOafj1g)
  * :heavy_check_mark: run-kiali-e2e-tests [#[1302]](https://paste.fedoraproject.org/paste/W4xgAWHuH~e8e9tu2odjOw)

**Comment by jotak** [comment]
@jmazzitelli I did the changes, and also fixed a couple of issues from a recently merged PR.
Tested myself both k8s and openshift scripts with and without local files, but more test from QE is appreciated :)

**Comment by rhqci** [comment]
Jenkins CI: kiali-core-pr-e2e-test [#601](https://paste.fedoraproject.org/paste/QCh-DzGy5Rhlb4-hL5UNgA)
  * :heavy_check_mark: run-kiali-e2e-tests [#[1305]](https://paste.fedoraproject.org/paste/iPit-gpkFMzezGSlN9BdbQ)

**Comment by rhqci** [comment]
Jenkins CI: kiali-core-pr-e2e-test [#600](https://paste.fedoraproject.org/paste/eE9S8bsRfu2vkNtjLIVH3g)
  * :heavy_check_mark: run-kiali-e2e-tests [#[1306]](https://paste.fedoraproject.org/paste/qBJwCQEaopiV-9mfWjpiCQ)

**Pull request merged**